node/external/logs/logsRepository.go

```diff
@@ -24,7 +24,23 @@ func newLogsRepository(storageService dataRetriever.StorageService, marshaller m
 	}
 }
 
+// getLog loads event & logs on a best-effort basis.
+// It searches for logs both in epoch N and epoch N-1, in order to overcome epoch-changing edge-cases.
 func (repository *logsRepository) getLog(logKey []byte, epoch uint32) (*transaction.Log, error) {
+	txLog, err := repository.doGetLog(logKey, epoch)
+	if err != nil {
+		txLog, errPreviousEpoch := repository.doGetLog(logKey, epoch-1)
```

Comment:
might check that epoch is not 0 before subtracting 1 from it

Reply:
Good catch :rocket: Fixed.